src/examples/pongxr/ball.js

```diff
@@ -0,0 +1,91 @@
+import State from "../../engine/state"
+import { SphereBufferGeometry, ShaderMaterial, PositionalAudio, AudioLoader, Mesh, PointLight, BackSide } from "three";
+import Physics from "../../engine/physics/physics"
+import frictionlessMat from "./frictionlessMaterial"
+
+const hitAudioFile = require("./assets/audio/elecping.ogg");
+
+const vs = require('./assets/shaders/vs_defaultVertex.glsl');
+const fs_puddles = require('./assets/shaders/fs_puddles.glsl');
+
+class Ball
+{
+    constructor(position, addRigidBody)
+    {
+        const ball = new Mesh(new SphereBufferGeometry(.2, 13, 13), new ShaderMaterial({ uniforms: { time: { value: 0.0 } }, vertexShader: vs, fragmentShader: fs_puddles }));
+        ball.position.copy(position);
+        ball.name = "ball";
+        this.initPos = position;
+
+        // physics 
+        if (addRigidBody == true)
+        {
+            // console.log("adding RB to Ball");
+            ball.rb = Physics.addRigidBody(ball, Physics.RigidBodyShape.Sphere, Physics.Body.DYNAMIC, 1);
+
+            ball.rb.material = frictionlessMat;
+            setTimeout(function ()
+            {
+                // Audiolistener setTimeOut hack
+                // it doesn't exist until after scene load time, so...
+                // (also see src/engine/engine.js)
+                // TODO: solve more cleanly.
```

Comment:
tbh I haven't mentally worked through this yet. Possible options seem:
- leave State out of Ball altogether (already a win!), leave the AudioListener out of State.globals (same) and simply parent the camera to it, then grab camera from `engine.js` and refer to its audiolistener directly - I think it would work?
- Catching a custom event when AudioListener is ready and hooking _all audio across the app_ up when it's fired? feels super clunky, right?

Going to try the first solution.

Reply:
Update: working, updated the PR (also for the audio in Level.js). This is _so_ much cleaner thanks to grabbing the listener from Camera. Thank you for this! (And everything else!)

Reply:
Whooooo!